app/components/Common/MenuOptions/index.tsx

```diff
@@ -0,0 +1,93 @@
+import i18next from 'i18next';
+import React, { useRef } from 'react';
+import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+import Menu, { MenuItem } from 'react-native-material-menu';
+import { Colors, Typography } from '../../../styles';
+import IconSwitcher from '../IconSwitcher';
+
+export type OptionsType = {
+  key: string;
+  disabled?: boolean;
+  iconType: string;
+  iconName: string;
+  text: string;
+  iconStyle?: any;
+};
+
+interface IMenuOptionsProps {
+  options: OptionsType[];
+  onOptionPress?: any;
+  containerStyle?: any;
+}
+
+const MenuOptions = ({ options, containerStyle, onOptionPress }: IMenuOptionsProps) => {
+  const menuRef = useRef<any>();
+
+  const showMenu = () => {
+    if (menuRef?.current?.show) {
+      menuRef.current.show();
+    }
+  };
+
+  const hideMenu = () => {
+    if (menuRef?.current?.hide) {
+      menuRef.current.hide();
+    }
+  };
+
+  return (
+    <View style={[containerStyle]}>
+      <Menu
+        ref={menuRef}
+        button={
+          <TouchableOpacity onPress={showMenu} style={{ padding: 10 }}>
+            <IconSwitcher
+              iconType={'MCIIcon'}
+              name={'dots-vertical'}
+              size={24}
+              color={Colors.TEXT_COLOR}
+            />
+          </TouchableOpacity>
+        }>
+        {options.map((option: OptionsType) => (
+          <MenuItem
+            key={option.key}
+            disabled={option.disabled}
+            style={styles.option}
+            onPress={() => {
+              onOptionPress(option);
+              hideMenu();
```

Comment:
Once you hide the menu also all opened dialog, e.g. the share dialog, get hidden as well! How can we solve this?

Reply:
Related to https://github.com/Plant-for-the-Planet-org/treemapper/pull/327#issuecomment-852849981 and https://github.com/Plant-for-the-Planet-org/treemapper/pull/327#issuecomment-853642036

Reply:
I just tried to trigger the `onOptionPress` from the `onHidden` callback in the menu component of https://www.npmjs.com/package/react-native-material-menu (storing the last selected menu items in a local state), but the problem persist that the share dialog gets hidden again :-(